### ISSUE TITLE
MuxStatsExoPlayer callback NPE crash

### ISSUE THREAD

**Issue by XyrizTan** [muxinc/mux-stats-sdk-exoplayer#113]
Hello, been encountering a null pointer exception crash in your latest 2.13.1/2.4.9 aar of your SDK

```
2021-06-09 12:19:11.988 25266-25266/com.viki.android E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.viki.android, PID: 25266
    java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mux.stats.sdk.core.model.a.a(java.lang.Long)' on a null object reference
        at com.mux.stats.sdk.muxstats.e$c.a(Unknown Source:12)
        at com.mux.stats.sdk.muxstats.e$e.a(Unknown Source:0)
        at com.mux.stats.sdk.muxstats.e$d.a(Unknown Source:29)
        at com.mux.stats.sdk.muxstats.MuxStatsExoPlayer.onLoadCompleted(Unknown Source:16)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.lambda$onLoadCompleted$28(AnalyticsCollector.java:503)
        at com.google.android.exoplayer2.analytics.-$$Lambda$AnalyticsCollector$LCUClj88hG_1RJ03wPFhEhogQu4.invoke(Unknown Source:8)
        at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.invoke(ListenerSet.java:297)
        at com.google.android.exoplayer2.util.ListenerSet.lambda$queueEvent$0(ListenerSet.java:184)
        at com.google.android.exoplayer2.util.-$$Lambda$ListenerSet$NbKDn9xtItiyMgYZmjIx_Sv1FFQ.run(Unknown Source:6)
        at com.google.android.exoplayer2.util.ListenerSet.flushEvents(ListenerSet.java:205)
        at com.google.android.exoplayer2.util.ListenerSet.sendEvent(ListenerSet.java:220)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.sendEvent(AnalyticsCollector.java:797)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.onLoadCompleted(AnalyticsCollector.java:500)
        at com.google.android.exoplayer2.source.MediaSourceEventListener$EventDispatcher.lambda$loadCompleted$1$MediaSourceEventListener$EventDispatcher(MediaSourceEventListener.java:301)
        at com.google.android.exoplayer2.source.-$$Lambda$MediaSourceEventListener$EventDispatcher$ZuTEoXsbK3i0dfIxRwW-S0zQ6UA.run(Unknown Source:8)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

Dependencies used:
Mux SDK aar v2.13.1 (v2.4.9)
Exoplayer v2.13.3
IMA SDK v3.24.0 (we depend on IMA SDK in our project)

Environment:
Android API 30 Emulator

This doesn't happen in your previous version Mux SDK aar v2.13.1 (v2.4.8) combined with IMA SDK 3.19.4 (3.22.3 up to 3.24.0 still crashes, only the mux 2.4.8 and IMA 3.19.4 combination seems to be stable). Kindly advise if this indeed is an issue on your end / a fix will be in the works. Thanks!

~Edit: Note We depend on IMA SDK in our own project so it might have an effect on the dependency resolution, going to try matching it with what mux SDK uses~ IMA 3.9.0 is too far back and we are already using most of their newer apis on certain classes

**Comment by jsanford8** [comment]
Hi - thanks for the report. We are looking at this right now.

**Comment by DeveloperYatin** [comment]
@jsanford8 i also got same error may be this logs will helps you 

> `2021-06-24 15:56:29.690 20353-20353/com.google.android.exoplayer2.demo E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.google.android.exoplayer2.demo, PID: 20353
    java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mux.stats.sdk.core.model.BandwidthMetricData.setRequestBytesLoaded(java.lang.Long)' on a null object reference
        at com.mux.stats.sdk.muxstats.MuxBaseExoPlayer$BandwidthMetric.onLoadCompleted(MuxBaseExoPlayer.java:864)
        at com.mux.stats.sdk.muxstats.MuxBaseExoPlayer$BandwidthMetricHls.onLoadCompleted(MuxBaseExoPlayer.java:904)
        at com.mux.stats.sdk.muxstats.MuxBaseExoPlayer$BandwidthMetricDispatcher.onLoadCompleted(MuxBaseExoPlayer.java:963)
        at com.mux.stats.sdk.muxstats.MuxStatsExoPlayer.onLoadCompleted(MuxStatsExoPlayer.java:175)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.onLoadCompleted(AnalyticsCollector.java:377)
        at com.google.android.exoplayer2.source.MediaSourceEventListener$EventDispatcher.lambda$loadCompleted$3$MediaSourceEventListener$EventDispatcher(MediaSourceEventListener.java:499)
        at com.google.android.exoplayer2.source.-$$Lambda$MediaSourceEventListener$EventDispatcher$lf0qUEczRZoeee9GZpzu7zKv-HE.run(Unknown Source:8)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:239)
        at android.app.ActivityThread.main(ActivityThread.java:8102)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:626)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1015)`

**Comment by skidder** [comment]
@XyrizTan @DeveloperYatin we fixed this issue in [v2.4.10](https://github.com/muxinc/mux-stats-sdk-exoplayer/releases/tag/v2.4.10) of our SDK and recommend trying the latest SDK (currently [v2.4.12](https://github.com/muxinc/mux-stats-sdk-exoplayer/releases/tag/v2.4.12)). Thanks again for the bug report!
